src/utils/proxy.js

```diff
@@ -320,12 +320,10 @@ const initializeProxy = async function (port, distDir, projectDir) {
     }
 
     const requestURL = new URL(req.url, `http://${req.headers.host || 'localhost'}`)
-    const pathHeaderRules = headersForPath(headerRules, requestURL.pathname)
-    if (!isEmpty(pathHeaderRules)) {
```

Comment:
This is useless since `forEach()` is a noop when `pathHeaderRules` is an empty object.